engine.py

```diff
@@ -473,19 +473,20 @@ def save(self, document):
         """
         Save the document in place
         """
+        # since Photoshop 24.1.0, saving an already saved file triggers errors
+        if not document.saved:
```

Comment:
Would it make sense to make this test right before `document.save()` instead of the full block?

Otherwise, can you minimize the change with:
```python
if document.saved:
    # since Photoshop 24.1.0, saving an already saved file triggers errors
    return
```